src/Fixer/Naming/HomoglyphNamesFixer.php

```diff
@@ -0,0 +1,250 @@
+        'Ｕ' => 'U',
+        'Ѵ' => 'V',
+        'Ⅴ' => 'V',
+        'Ｖ' => 'V',
+        'Ｗ' => 'W',
+        'Χ' => 'X',
+        'Х' => 'X',
+        'Ⅹ' => 'X',
+        'Ｘ' => 'X',
+        'ʏ' => 'Y',
+        'Υ' => 'Y',
+        'Ү' => 'Y',
+        'Ｙ' => 'Y',
+        'Ζ' => 'Z',
+        'Ｚ' => 'Z',
+        'ɑ' => 'a',
+        'а' => 'a',
+        'ａ' => 'a',
+        'Ь' => 'b',
+        'ｂ' => 'b',
+        'ϲ' => 'c',
+        'с' => 'c',
+        'ⅽ' => 'c',
+        'ｃ' => 'c',
+        'ԁ' => 'd',
+        'ⅾ' => 'd',
+        'ｄ' => 'd',
+        'е' => 'e',
+        'ｅ' => 'e',
+        'ｆ' => 'f',
+        'ɡ' => 'g',
+        'ｇ' => 'g',
+        'һ' => 'h',
+        'ｈ' => 'h',
+        'ɩ' => 'i',
+        'і' => 'i',
+        'ⅰ' => 'i',
+        'ｉ' => 'i',
+        'ј' => 'j',
+        'ｊ' => 'j',
+        'ｋ' => 'k',
+        'ⅼ' => 'l',
+        'ｌ' => 'l',
+        'ⅿ' => 'm',
+        'ｍ' => 'm',
+        'ｎ' => 'n',
+        'ο' => 'o',
+        'о' => 'o',
+        'ｏ' => 'o',
+        'р' => 'p',
+        'ｐ' => 'p',
+        'ｑ' => 'q',
+        'ｒ' => 'r',
+        'ѕ' => 's',
+        'ｓ' => 's',
+        'ｔ' => 't',
+        'ｕ' => 'u',
+        'ν' => 'v',
+        'ѵ' => 'v',
+        'ⅴ' => 'v',
+        'ｖ' => 'v',
+        'ѡ' => 'w',
+        'ｗ' => 'w',
+        'х' => 'x',
+        'ⅹ' => 'x',
+        'ｘ' => 'x',
+        'у' => 'y',
+        'ｙ' => 'y',
+        'ｚ' => 'z',
+    ];
+
+    /**
+     * Returns the definition of the fixer.
+     *
+     * @return FixerDefinitionInterface
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Replace accidental usage of non ascii characters in names.',
+            [new CodeSample('<?php $nаmе = \'wrong\';')],
+            null,
+            'Risky as it might rename classes that are autoloaded and cannot rename the files'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * Check if the fixer is a candidate for given Tokens collection.
+     *
+     * Fixer is a candidate when the collection contains tokens that may be fixed
+     * during fixer work. This could be considered as some kind of bloom filter.
+     * When this method returns true then to the Tokens collection may or may not
+     * need a fixing, but when this method returns false then the Tokens collection
+     * need no fixing for sure.
+     *
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_VARIABLE, T_STRING]);
+    }
+
+    public static function match($a)
+    {
+        return static::$replacements[$a];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens->findGivenKind([T_VARIABLE, T_STRING]) as $tokensByKind) {
```

Comment:
As far as I could tell, these are all the tokens that are literal names of classes etc. Its a valid use case to have cyrillic or other symbols in strings used by code.